main.nf

```diff
@@ -146,6 +148,7 @@ process list_objects {
     -e 'synapseConfig' -e 'synapse_config' \
   | awk '{\$1=\$2=\$3=""; print \$0}' \
   | sed 's|^   |s3://${bucket}/|' \
+  ${filename_string ? "| grep '${filename_string}'" : ""} \
```

Comment:
This does require the entire S3 prefix to be listed, which might not be the fastest approach (could consider using s3 ls with --exclude and --include flags instead). However, I like the simplicity of it and it worked fine with a prefix that contained over 300,000 files when I tested it.

Reply:
Thanks @wpoehlm, can you provide an example usage of this?

Reply:
@thomasyu888 sure thing, to add to the existing example in the readme your parameters would look like this:

s3_prefix: "s3://example-bucket/outputs/"
parent_id: "syn26601236"
filename_string: "g.vcf"

In this example, only files with 'g.vcf' in the filename would be indexed. Let me know if you'd like me to add an example like this to the readme.

Reply:
Thanks @wpoehlm , sorry for the delay in response.  That would be great to add an example in the readme.  I will pre-approve this PR.

Reply:
@thomasyu888 done, thanks!